testapi.pm

```diff
@@ -1144,22 +1144,28 @@ sub release_key {
 
 =head2 send_key_until_needlematch
 
-  send_key_until_needlematch($tag, $key [, $counter, $timeout]);
+  send_key_until_needlematch($tag, $key [, $counter, $timeout, $no_wait]);
 
 Send specific key until needle with C<$tag> is not matched or C<$counter> is 0.
-C<$tag> can be string or C<ARRAYREF> (C<['tag1', 'tag2']>)
-Default counter is 20 steps, default timeout is 1s
+C<$tag> can be string or C<ARRAYREF> (C<['tag1', 'tag2']>).
+Default counter is 20 steps, default timeout is 1s.
+
+Specify C<$no_wait> to run the screen check as fast as possible that is
+possibly more than once per second which is default. Select this to check a
+screen which can change in a range faster than 1-2 seconds not to miss the
+screen to check for.
 
 Throws C<FailedNeedle> exception if needle is not matched until C<$counter> is 0.
 
 =cut
 
 sub send_key_until_needlematch {
-    my ($tag, $key, $counter, $timeout) = @_;
+    my ($tag, $key, $counter, $timeout, $no_wait) = @_;
```

Comment:
Actually what you could do is just consume all additional parameters inside %args including $timeout and forward that one to the internal `check_screen` call

Reply:
But wouldn't that suggestion break the existing API? E.g. `send_key_until_needlematch('foo', 'bar', 5, 20)` would then need to be adjusted to `send_key_until_needlematch('foo', 'bar', counter => 5, timeout => 20)`.

Reply:
I think that compatibility can be kept like in `check_screen` or `assert_screen`. I will try this.

Reply:
I change the code to be able to use `%args` syntax with additional parameters and this code also keep the old API.

I tested both calls in this run (test failure is not because of this change): http://1b147.qa.suse.de/tests/4105